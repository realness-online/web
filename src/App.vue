<style src="@/style/index.styl" lang="stylus"></style>
<template lang="html">
  <main id="realness">
    <h6>Version {{version}}</h6>
    <router-view></router-view>
    <developer-tools></developer-tools>
  </main>
</template>
<script>
  import developer_tools from '@/components/developer'
  import * as firebase from 'firebase/app'
  export default {
    watch: {
      '$route' (to, from) {
        sessionStorage.previous = from.path
      }
    },
    data() {
      return {
        version: process.env.VUE_APP_VERSION
      }
    },
    created() {
      console.log('initialize app')
      let config = {
        apiKey: process.env.VUE_APP_API_KEY,
        authDomain: process.env.VUE_APP_AUTH_DOMAIN,
        databaseUrl: process.env.VUE_APP_DATABASE_URL,
        projectId: process.env.VUE_APP_PROJECT_ID,
        storageBucket: process.env.VUE_APP_STORAGE_BUCKET,
        messagingSenderId: process.env.VUE_APP_MESSAGING_SENDER_ID
      }
      firebase.initializeApp(config)
    },
    components: {
      'developer-tools': developer_tools
    }
  }
</script>
<style lang="stylus">
  @require '/style/variables'
  #realness > h6
    position:fixed
    bottom: base-line
    left: base-line
</style>
