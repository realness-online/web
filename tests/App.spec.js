import { shallowMount } from '@vue/test-utils'
import App from '@/App'
describe('@/App.vue', () => {
  let wrapper
  const node_env = import.meta.env
  beforeEach(async () => {
    process.env = { ...node_env }
    wrapper = await shallowMount(App, {
      global: {
        stubs: ["router-link", "router-view"]
      }
    })
  })
  afterEach(() => {
    wrapper.unmount()
    // vi.resetModules()
  })
  afterAll(() => {
    process.env = node_env
  })
  describe('Renders', () => {
    it('Layout of the application', () => {
      expect(wrapper.element).toMatchSnapshot()
    })
  })
  describe('Methods', () => {
    describe('#onLine', () => {
      it('Turns the editable content back on', () => {
        const elements = [{ setAttribute: vi.fn() }]

        wrapper.vm.status = 'offline'
        vi.spyOn(document, 'querySelectorAll').mockReturnValueOnce(elements)
        wrapper.vm.online()

        expect(wrapper.vm.status).toBe(null)
        expect(elements[0].setAttribute).toBeCalled()
      })
    })
    describe('#offLine', () => {
      it('Turns the editable content back on', () => {
        const elements = [{ setAttribute: vi.fn() }]
        wrapper.vm.status = 'offline'
        vi.spyOn(document, 'querySelectorAll').mockReturnValueOnce(elements)
        wrapper.vm.offline()
        expect(wrapper.vm.status).toBe('offline')
        expect(elements[0].setAttribute).toBeCalled()
      })
    })
    describe('#sync_active', () => {
      it('Sets status to active when syncing', () => {
        wrapper.vm.status = 'offline'
        wrapper.vm.sync_active(true)
        expect(wrapper.vm.status).toBe('working')
      })
      it('Sets status to null when not syncing', () => {
        wrapper.vm.status = 'offline'
        wrapper.vm.sync_active(false)
        expect(wrapper.vm.status).toBe(null)
      })
    })
  })
})
