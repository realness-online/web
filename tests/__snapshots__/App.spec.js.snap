// Vitest Snapshot v1

exports[`@/App.vue > Renders > Layout of the application 1`] = `
<main
  class=""
  id="realness"
>
  <router-view-stub />
  <aside>
    <sync-stub />
  </aside>
</main>
`;
